comp/forwarder/orchestrator/component.go

```diff
@@ -0,0 +1,20 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+// Package orchestrator implements the orchestrator forwarder component.
+package orchestrator
+
+import "github.com/DataDog/datadog-agent/comp/forwarder/defaultforwarder"
+
+// team: agent-metrics-logs
+
+// Component is the component type.
+type Component interface {
```

Comment:
Why is this component mimic the `optionnal.Option` interface rather than exposing the method of the orchestrator ? This doesn't really extract the orchestrator as a component.

I think I'm missing the pros of receiving as an argument a `orchestratorForwarder.Component` instead of `option.Optional[defaultforwarder.Forwarder]`.

If we don't want to migrate the orchestrator yet we might want to create a true wrapper above it like we did for config. That way the code using this component will not change when we actually migrate the orchestrator.

Reply:
I think my point was more: why have a component that act like a getter/factory instead of directly offering the final methods ?

ie, instead of:
```
orchestratorForwarder, found := orchestratorComp.Get()
if !found {
	return errors.New("orchestrator forwarder is not setup")
}
responses, err := orchestratorForwarder.SubmitOrchestratorChecks([...])
```
Why doesn't the component behave like:
```
responses, err := orchestratorComp.SubmitOrchestratorChecks([...])
```

In other word we don't have an orchestrator component but a forwarder getter, which doesn't achieve the goal of the component framework, I think.

Am I missing something ?

Reply:
The reason is just that orchestrator is optional at least until we can unify `nil/none` and `noopforwarder`. 
It is really the same as having `optional[Component]` but without the drawback explains in my previous comment.